src/test/java/com/exactpro/th2/crawler/CrawlerTest.java

```diff
@@ -249,7 +260,14 @@ public void handshakeNeededAnother() throws IOException, UnexpectedDataProcessor
             return EventResponse.newBuilder().setId(eventID).setStatus(Status.newBuilder().setHandshakeRequired(true).build()).build();
         });
 
-        Assertions.assertThrows(UnexpectedDataProcessorException.class, crawler::process);
+        when(manager.getDataServiceMock().intervalStart(any(IntervalInfo.class))).thenReturn(
+                IntervalStartResponse.newBuilder().setStatus(Status.newBuilder().setHandshakeRequired(true).build()).build()
+        );
+
+        crawler.process();
+
+        Assertions.assertEquals(dataProcessorName, crawler.getProcessorInfo().getDataProcessorName());
+        Assertions.assertEquals(dataProcessorVersion, crawler.getProcessorInfo().getDataProcessorVersion());
```

Comment:
Please extract to separate test

Reply:
There used to be `assertThrows` check, now there's no exception in Crawler in this case. So I just assert a correct name and version. What exactly do you want me to extract to a separate test?